### PR TITLE
Added support for automatically enabling the required RHEL7, OL7 and SL7 repos

### PR DESCRIPTION
**Pull request by jonnytpuppet**
Added support for automatically enabling the required RHEL7, OL7 and SL7 repos when installing docker.

### CODE REVIEW

File: manifests/install.pp [hunk 1]
```diff
@@ -107,9 +107,17 @@
...
 
   if $docker::manage_package {
-    package { 'docker':
-      ensure => $docker::ensure,
-      name   => $dockerpackage,
+    if $docker::repo_opt {
+      package { 'docker':
+          ensure          => $docker::ensure,
+          name            => $dockerpackage,
+          install_options => $docker::repo_opt,
```
Comment (garethr) [inline manifests/install.pp]: Isn't that going to be in the params scope? ie. `$docker::params::repo_opt`?

File: manifests/install.pp [hunk 2]
```diff
@@ -107,9 +107,17 @@
   }
 
   if $docker::manage_package {
-    package { 'docker':
-      ensure => $docker::ensure,
-      name   => $dockerpackage,
+    if $docker::repo_opt {
+      package { 'docker':
+          ensure          => $docker::ensure,
```
Comment (garethr) [inline manifests/install.pp]: Whitespace issue for this package resource

File: manifests/install.pp (hunk 1 above)
Comment (jonnytpuppet) [inline manifests/install.pp]: They both work as docker inherits docker::params. Which is preferred?

**Comment by garethr** [comment]
Could you add a simple unit test to validate this is going the right thing? ie. install_options isn;t set of some flavour of Debian and is set correctly for some flavour of RedHat.

**Pull request merged**